### ISSUE TITLE
List JWT Keys from JWTAuthorities of the common Bundle do grow from restart to restart without any limit

### ISSUE THREAD

**Issue by nicolaenicora** [spiffe/spire#4429]
Performance downgrading on calling `GeBundle` API  from the spire server due to accumulations of JWT Keys on the bundle.

---
 
**Observation:**
On the server configuration, we do have the following

`ca_ttl = "xxxxxh" // translated in days can be xx days for example`

Spire server does use this attribute configuration `ca_ttl` for Intermediate CA certificates (passed to the Upstream Authority) and JWT Keys as expiration time.
On each server start/restart a new JWT Key is generated and appended to the common bundle.
Having N replicas of the spire server and M restarts for each replica we do end up having N*M JWT Keys on the common bundle, for the configured period `ca_ttl` + 24h (which is hardcoded as a grace time for removing old/expired keys).
We noticed huge performance downgrades due to this growing list of JWT keys when the GetBundle API is called by the Spire agent.

**Expectation:**
We do expect the Spire server to use the `ca_ttl` configuration just for Intermediate CA certificates as expiration time, NOT for JWT Key.
For the JWT Key separate TTL configuration to be introduced `jwt_ttl`.
`
jwt_ttl = "xxxxh" // used to calculate the JWT Key expiration time
`

In our use case we don't need JWT Authorities at all, do propose to have the possibility to disable the publishing of the JWT Keys using a configuration flag.
`
jwt_publishing_enabled = true // by default enabled
`

* **Version**: 1.5.x  (most probably any versions)
* **Platform**:  any
* **Subsystem**:  Spire Server

**Comment by azdagron** [comment]
> On each server start/restart a new JWT Key is generated and appended to the common bundle.

What KeyManager have you configured on the server? If it is not a persisting Key Manager then SPIRE has no choice but to rotate into a new key (for both the X.509 and JWT authority) on restart...

**Comment by nicolaenicora** [comment]
> What KeyManager have you configured on the server? If it is not a persisting Key Manager then SPIRE has no choice but to rotate into a new key (for both the X.509 and JWT authority) on restart...

Yes, I do use the memory KeyManager. In any way, ...,  would be good as described early to have a separate ttl for JWT keys and a way to disable the publishing of the key to the bundle.

**Comment by azdagron** [comment]
What you observed is in line with the use of a memory KeyManager. Without persisted keys, the server will regenerate a new signing key for both X.509 and JWT on each restart.

Disabling JWT functionality in SPIRE Server may have larger implications. Give me some time to think those through.

**Comment by amartinezfayo** [comment]
We discussed this during the last maintainer's sync and we agreed that being able to disable JWT-SVID support in SPIRE would be a good thing to have. There are still some questions that we need to answer first in order to be able to start an implementation to allow disabling JWT-SVID support. I've filed #4473 to 1) capture this feature request and 2) scope the work needed.

Closing this in favor of #4473.
